Building master is broken by 2a068a1394d179dda4becf350e3afb4e8819675e

### System information
<!--  add version after "|" character -->
Type | Version/Name
 --- | ---
Distribution Name	|Gentoo
Distribution Version	|2.9
Kernel Version	|5.16.14-gentoo-x86_64
Architecture	|x86_64
OpenZFS Version	|master
<!--
Command to find OpenZFS version:
zfs version

Commands to find kernel version:
uname -r		# Linux
freebsd-version -r	# FreeBSD
-->

### Describe the problem you're observing
Building master fails with:
```
tests/zfs-tests/cmd/idmap_util.c:103:8: error: redefinition of ‘struct mount_attr’
  103 | struct mount_attr {
      |        ^~~~~~~~~~
In file included from /usr/include/sys/mount.h:32,
                 from ./lib/libspl/include/os/linux/sys/mount.h:27,
                 from ./lib/libspl/include/os/linux/sys/stat.h:31,
                 from tests/zfs-tests/cmd/idmap_util.c:33:
/usr/include/linux/mount.h:129:8: note: originally defined here
  129 | struct mount_attr {
      |        ^~~~~~~~~~
```
### Describe how to reproduce the problem
```
git clone https://github.com/openzfs/zfs.git
cd zfs
git clean -xdf && ./autogen.sh && ./configure --enable-debug --enable-debuginfo --with-config=all && make -j$(nproc)
```

2a068a1394d179dda4becf350e3afb4e8819675e introduced this code.

The problem code is:
```
#ifndef mount_attr
struct mount_attr {
        __u64 attr_set;
        __u64 attr_clr;
        __u64 propagation;
        __u64 userns_fd;
};
#endif
```
I am not sure how this was expected to work in the first place. You cannot use the CPP to check what C structures are defined. An autotools check is needed for that.

--- a/tests/zfs-tests/cmd/idmap_util.c
+++ b/tests/zfs-tests/cmd/idmap_util.c
@@ -99,18 +99,16 @@
 #define	AT_RECURSIVE 0x8000
 #endif
 
-#ifndef mount_attr
-struct mount_attr {
+typedef struct {
 	__u64 attr_set;
 	__u64 attr_clr;
 	__u64 propagation;
 	__u64 userns_fd;
-};
-#endif
+} mount_attr_t;
 
 static inline int
 sys_mount_setattr(int dfd, const char *path, unsigned int flags,
-    struct mount_attr *attr, size_t size)
+    mount_attr_t *attr, size_t size)
 {
 	return (syscall(__NR_mount_setattr, dfd, path, flags, attr, size));
 }
@@ -528,7 +526,7 @@ is_idmap_supported(char *path)
 	list_t head;
 	int ret;
 	int tree_fd = -EBADF, path_fd = -EBADF;
-	struct mount_attr attr = {
+	mount_attr_t attr = {
 	    .attr_set	= MOUNT_ATTR_IDMAP,
 	    .userns_fd  = -EBADF,
 	};
@@ -634,7 +632,7 @@ do_idmap_mount(list_t *idmap, char *source, char *target, int flags)
 {
 	int ret;
 	int tree_fd = -EBADF, source_fd = -EBADF;
-	struct mount_attr attr = {
+	mount_attr_t attr = {
 	    .attr_set   = MOUNT_ATTR_IDMAP,
 	    .userns_fd  = -EBADF,
 	};

